src/abstractdataframe/iteration.jl

```diff
@@ -75,12 +75,16 @@ x  3
 eachrow(df::AbstractDataFrame) = DataFrameRows(df, index(df))
 
 Base.IndexStyle(::Type{<:DataFrameRows}) = Base.IndexLinear()
-Base.size(itr::DataFrameRows) = (size(itr.df, 1), )
+Base.size(itr::DataFrameRows) = (size(parent(itr), 1), )
 
 Base.@propagate_inbounds Base.getindex(itr::DataFrameRows, i::Int) =
-    DataFrameRow(itr.df, itr.index, i)
+    DataFrameRow(parent(itr), getfield(itr, :index), i)
```

Comment:
Define an `index` method for `DataFrameRows`?

Reply:
Actually maybe we should remove the `:index` field. We currently do not use it. It is always set to `index(df)` and we do not export `index` anyway.

Reply:
OK, simplification is always nice.

Reply:
I have changed this - let us see how it goes.